sklearn/utils/tests/test_validation.py

```diff
@@ -992,12 +993,21 @@ class WrongDummyMemory:
     pass
 
 
-@pytest.mark.filterwarnings("ignore:The 'cachedir' attribute")
 def test_check_memory():
+    joblib_before_0_12 = parse_version(joblib.__version__) < parse_version("0.12")
+
     memory = check_memory("cache_directory")
-    assert memory.cachedir == os.path.join("cache_directory", "joblib")
+    if joblib_before_0_12:
+        assert memory.cachedir == os.path.join("cache_directory", "joblib")
```

Comment:
We only test the minimum version of joblib in the `ubuntu_altas` job and coverage is turned off there.